src/node/SetupServerApi.ts

```diff
@@ -48,14 +49,23 @@ export class SetupServerApi
     this.init()
   }
 
-  private createContext(): SetupServerInternalContext {
-    return {
-      get nodeEvents() {
-        return import('node:events')
-          .then((events) => events)
-          .catch(() => undefined)
-      },
-    }
+  /**
+   * Appends a context onto the SetupServerApi instance,
+   * returning that instance. This is used to provide
+   * a dependency injection mechanism for the Node-specific
+   * modules.
+   *
+   * In a future major we might want to consider adding a
+   * new argument to the constructor to allow for this
+   * as a bit more streamlined setup, but this allows
+   * us to remove a dynamic `node:events` import, provide
+   * the same functionality, and remove a jest/jsdom/node bug
+   * where dynamically importing a node module could segfault
+   *
+   */
+  withContext(context: SetupServerContext) {
+    this.context = context
```

Comment:
this is a bit unusual. the concept of context isn't strictly defined in MSW internally but the way we use it in the worker api is that api creates the context within. with this change, we now have setupworker that has internal, non-extendable context, and setupserver that exclusively relies on the context being provided by the consumer. i think that's a mix of intentions and we probably should drop the word "context" to prevent any confusion. wdyt about `.withModules()` or something? and use that exclusively for standard modules injection.

Reply:
@mattcosta7 i also think we should make this method private and call it explicitly internally. i'm not comfortable with this api enough to make it public and thus prevent us from iterating on it in the future. this isn't something MSW users need anyway.